### PR TITLE
colorschemes/base16: use mkVimPlugin

### PR DESCRIPTION
**Pull request by GaetanLepage**
This is, in my honest opinion, a way nicer module design to handle this plugin.

### CODE REVIEW

File: plugins/colorschemes/base16/default.nix [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    name = "base16";
+    isColorscheme = true;
+    originalName = "base16.nvim";
+    defaultPackage = pkgs.vimPlugins.base16-nvim;
+
+    maintainers = [maintainers.GaetanLepage];
+
+    imports = [
+      (
+        mkRemovedOptionModule
```
Comment (traxys) [inline plugins/colorschemes/base16/default.nix:21]: Can't we use a mkRenamedOption?

File: plugins/colorschemes/base16/default.nix [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        then {
+          colorscheme = "base16-${cfg.colorscheme}";
+        }
+        # ...otherwise, call the `setup function`.
+        else {
+          # Explicitly set the `colorscheme` option to `null` as it has been set to
+          # `mkDefault "base16"` by `mkVimPlugin`.
+          colorscheme = null;
+          extraConfigLuaPre = ''
+            require('base16-colorscheme').setup(${helpers.toLuaObject cfg.customColorScheme})
```
Comment (traxys) [inline plugins/colorschemes/base16/default.nix:114]: Shouldn't this be cfg.colorscheme?

File: plugins/colorschemes/base16/default.nix [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    imports = [
+      (
+        mkRemovedOptionModule
+        ["colorschemes" "base16" "customColorScheme"]
+        "You can directly provide an attrs to the `colorschemes.base16.colorscheme` option."
+      )
+    ];
+
+    extraOptions = {
+      useTruecolor = mkOption {
```
Comment (traxys) [inline plugins/colorschemes/base16/default.nix:28]: This option seems pretty useless as it simply maps to a builtin neovim option. Shouldn't we deprecate it with a mkRenameOption to the correct neovim builtin option?

**Review by traxys — CHANGES_REQUESTED** [review]

File: plugins/colorschemes/base16/default.nix (hunk 3 above)
Comment (GaetanLepage) [reply plugins/colorschemes/base16/default.nix:31]: Sure.

File: plugins/colorschemes/base16/default.nix (hunk 1 above)
Comment (GaetanLepage) [reply plugins/colorschemes/base16/default.nix]: Indeed, good idea !

File: plugins/colorschemes/base16/default.nix [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    in [
+      (
+        mkRenamedOptionModule
+        (basePluginPath ++ ["customColorScheme"])
+        (basePluginPath ++ ["colorscheme"])
+      )
+      (
+        mkRenamedOptionModule
+        (basePluginPath ++ ["useTruecolor"])
+        ["termguicolors"]
```
Comment (traxys) [inline plugins/colorschemes/base16/default.nix:34]: isn't this `["options" "termguicolors"]` ? Our very unfortunate mapping to neovim options...

**Review by traxys — CHANGES_REQUESTED** [review]

**Review by traxys — APPROVED** [review]

**Pull request merged**